### ISSUE TITLE
Is there any way to render a table inside a modal?

### ISSUE THREAD

**Issue by alash3al** [orchidsoftware/platform#1103]

**Comment by alash3al** [comment]
@tabuna fine, but what if I want to pass parameters to the table instead of the fixed data source, i.e, when a user clicked on `ModalToggle`, I want to pass a param containing the model ID, just like what we do in the asyncModalParams

**Comment by tabuna** [comment]
No problem, it's just as possible.
See an example of loading a similar window in the user list view screen.
https://github.com/orchidsoftware/platform/blob/dd7760921b8f9a11749d6b49f6ad7e7a028c3de7/install-stubs/Orchid/Layouts/User/UserListLayout.php#L42-L46

**Comment by alash3al** [comment]
I know that, I think I didn't describe it well, let me try again:

I want to show a table in a modal, but with a dynamic source, i.e:

the user went to posts listView, then he clicked on a post history link, then I want a popup to display a table of historical changes of that post

**Comment by tabuna** [comment]
Of course, there is no example.
Here is an example with this:

```php
<?php

declare(strict_types=1);

namespace App\Orchid\Screens;

use Orchid\Screen\Actions\ModalToggle;
use Orchid\Screen\Layout;
use Orchid\Screen\Repository;
use Orchid\Screen\Screen;
use Orchid\Screen\TD;

class PlatformScreen extends Screen
{
    /**
     * Display header name.
     *
     * @var string
     */
    public $name = 'Table in modal';

    /**
     * Display header description.
     *
     * @var string
     */
    public $description;

    /**
     * Query data.
     *
     * @return array
     */
    public function query(): array
    {
        return [
            'example-table' => [],
            'posts' => [
                new Repository(['id' => 1, 'price' => 10.24, 'created_at' => '01.01.2020']),
                new Repository(['id' => 2, 'price' => 65.9, 'created_at' => '01.01.2020']),
            ],
        ];
    }

    /**
     * Button commands.
     *
     * @return Action[]
     */
    public function commandBar(): array
    {
        return [
            ModalToggle::make('Launch modal')
                ->modal('table-in-modal')
                ->method('query')
                ->icon('icon-full-screen'),
        ];
    }


    public function asyncLoadHistory(int $id)
    {
        // Here you should load the data for your table

        if($id == 1){
            $query =  new Repository(['id' => 1, 'description' => 'Change FOO', 'created_at' => '01.01.2020']);
        }else{
            $query =  new Repository(['id' => 2, 'description' => 'Change BAR', 'created_at' => '01.01.2020']);
        }

        return [
            'example-table' => [
                $query
            ],
        ];
    }

    /**
     * Views.
     *
     * @return Layout[]
     */
    public function layout(): array
    {
        return [

            Layout::table('posts',[
                TD::set('id', 'ID'),
                TD::set('price', 'Price'),
                TD::set('created_at', 'Created'),
                TD::set('email', __('Email'))
                    ->render(function (Repository $repository) {
                        return ModalToggle::make('Load history')
                            ->modal('table-in-modal')
                            ->modalTitle('History')
                            ->method('unknown')
                            ->asyncParameters($repository->get('id'));
                    }),
            ]),

            Layout::modal('table-in-modal', [
                Layout::table('example-table', [
                    TD::set('id', 'ID'),
                    TD::set('description', 'Description'),
                    TD::set('created_at', 'Created'),
                ])
            ])->async('asyncLoadHistory')
        ];
    }
}

```

**Comment by alash3al** [comment]
@tabuna Thank you very much for your help, I didn't that I must declare it as an empty array in the query, as it was giving me "Invalid argument supplied for foreach() "

**Comment by tabuna** [comment]
Even a hidden modal window renders content. Therefore, we must specify the default values. This works well when data is loading for a long time.

To avoid this, use the hide methods in the real table class:

```php
<?php

class HistoryTableLayout extends Table
{
    public function canSee(Repository $query): bool
    {
        return $query->has('history-table');
    }
}
```
